backend/director/agents/video_generation.py

```
@@ -136,6 +167,10 @@ async def run_async(
                 raise Exception(f"{engine} not supported")
 
             os.makedirs(DOWNLOADS_PATH, exist_ok=True)
+            if not os.access(DOWNLOADS_PATH, os.W_OK):
+                raise PermissionError(
+                    f"No write permission for output directory: {output_dir}"
```

Ruff (0.8.2)

Undefined name `output_dir`

(F821)